libs/tests/core/test_typeddict_output_execution.py

```diff
@@ -0,0 +1,244 @@
+"""
+End-to-end tests for TypedDict output execution through the entire tool pipeline.
+"""
+
+from typing import Annotated, Optional
+
+import pytest
+from arcade_core.catalog import ToolCatalog, create_func_models
+from arcade_core.executor import ToolExecutor
+from arcade_core.schema import ToolContext
+from arcade_tdk import tool
+from typing_extensions import TypedDict
+
+
+# Define various TypedDict structures for testing
+class SimpleDict(TypedDict):
+    """A simple typed dictionary."""
+
+    name: str
+    value: int
+
+
+class NestedDict(TypedDict):
+    """A nested typed dictionary."""
+
+    id: int
+    info: SimpleDict
+    tags: list[str]
+
+
+class OptionalFieldsDict(TypedDict, total=False):
+    """TypedDict with optional fields."""
+
+    required_field: str
+    optional_field: int
+
+
+# Define test tools
+@tool
+def returns_typeddict() -> Annotated[SimpleDict, "Returns a simple TypedDict"]:
+    """Tool that returns a TypedDict."""
+    return SimpleDict(name="test", value=42)
+
+
+@tool
+def returns_list_of_typeddict() -> Annotated[list[SimpleDict], "Returns list of TypedDict"]:
+    """Tool that returns a list of TypedDict."""
+    return [
+        SimpleDict(name="item1", value=1),
+        SimpleDict(name="item2", value=2),
+        SimpleDict(name="item3", value=3),
+    ]
+
+
+@tool
+def returns_optional_typeddict(
+    return_none: Annotated[bool, "Whether to return None"] = False,
+) -> Annotated[Optional[SimpleDict], "Returns optional TypedDict"]:
+    """Tool that returns an optional TypedDict."""
+    if return_none:
+        return None
+    return SimpleDict(name="optional", value=100)
+
+
+@tool
+def returns_nested_typeddict() -> Annotated[NestedDict, "Returns nested TypedDict"]:
+    """Tool that returns a nested TypedDict."""
+    return NestedDict(id=1, info=SimpleDict(name="nested", value=99), tags=["tag1", "tag2", "tag3"])
+
+
+@tool
+def accepts_and_returns_typeddict(
+    data: Annotated[SimpleDict, "Input TypedDict"],
+) -> Annotated[SimpleDict, "Modified TypedDict"]:
+    """Tool that accepts and returns a TypedDict."""
+    return SimpleDict(name=f"Modified: {data['name']}", value=data["value"] * 2)
+
+
+@tool
+def returns_dict_list() -> Annotated[list[dict], "Returns list of dicts"]:
+    """Tool that returns a list of dictionaries including TypedDicts."""
+    return [
+        {"type": "plain", "value": 42},
+        {"name": "string", "data": "test"},
+        SimpleDict(name="typed", value=99),
+    ]
+
+
+class TestTypeDictOutputExecution:
+    """Test TypedDict outputs through the full execution pipeline."""
+
+    @pytest.fixture
+    def catalog(self):
+        return ToolCatalog()
+
+    @pytest.fixture
+    def context(self):
+        return ToolContext()
+
+    async def test_returns_typeddict(self, catalog, context):
```

Comment:
I think the Github action that tests libs is failing because you need to mark these async test functions with `@pytest.mark.asyncio`. For example [this executor test](https://github.com/ArcadeAI/arcade-ai/blob/7888dc505e13e2ac831a4f0dc3318486f70bb24d/libs/tests/core/test_executor.py#L59) uses that decorator @Spartee